components/canvas/raqote_backend.rs

```diff
@@ -527,11 +527,33 @@ impl GenericDrawTarget for raqote::DrawTarget {
         pattern: &canvas_data::Pattern,
         options: &DrawOptions,
     ) {
-        self.draw_text(
+        let mut start = pathfinder_geometry::vector::vec2f(start.x, start.y);
+        let mut ids = Vec::new();
+        let mut positions = Vec::new();
+        for c in text.chars() {
+            let id = match font.glyph_for_char(c) {
+                Some(id) => id,
+                None => {
+                    warn!("Skipping non-existent glyph {}", c);
+                    continue;
+                },
+            };
+            ids.push(id);
+            positions.push(Point2D::new(start.x(), start.y()));
+            let advance = match font.advance(id) {
+                Ok(advance) => advance,
+                Err(e) => {
+                    warn!("Skipping glyph {} with missing advance: {:?}", c, e);
+                    continue;
+                },
+            };
+            start += advance * point_size / 24. / 96.;
```

Comment:
Where do these values come from? Are you sure this isn't the `units_per_em` for some font?

Reply:
This is copied directly from the code we used to call in https://github.com/jrmuizel/raqote/blob/490bff22610a130ad75eb797b6bd6e3e361bbde5/src/draw_target.rs#L742-L761.